feature/bgp/policybase/otg_tests/route_installation_test/route_installation_test.go

```diff
@@ -325,123 +334,168 @@ func verifyPolicyTelemetry(t *testing.T, dut *ondatra.DUTDevice, policy string)
-func verifyTraffic(t *testing.T, ate *ondatra.ATEDevice, allFlows []*ondatra.Flow, wantLoss bool) {
-	captureTrafficStats(t, ate, wantLoss)
-	for _, flow := range allFlows {
-		lossPct := ate.Telemetry().Flow(flow.Name()).LossPct().Get(t)
+func verifyTraffic(t *testing.T, ate *ondatra.ATEDevice, c gosnappi.Config, wantLoss bool) {
+	otg := ate.OTG()
+	otgutils.LogFlowMetrics(t, otg, c)
+	for _, f := range c.Flows().Items() {
+		t.Logf("Verifying flow metrics for flow %s\n", f.Name())
+		recvMetric := otg.Telemetry().Flow(f.Name()).Get(t)
+		txPackets := recvMetric.GetCounters().GetOutPkts()
+		rxPackets := recvMetric.GetCounters().GetInPkts()
+		lostPackets := txPackets - rxPackets
+		lossPct := lostPackets * 100 / txPackets
 		if !wantLoss {
-			if lossPct > tolerancePct {
-				t.Errorf("Traffic Loss Pct for Flow: %s\n got %v, want 0", flow.Name(), lossPct)
+			if lostPackets > tolerance {
+				t.Logf("Packets received not matching packets sent. Sent: %v, Received: %d", txPackets, rxPackets)
+			}
+			if lossPct > tolerancePct && txPackets > 0 {
+				t.Errorf("Traffic Loss Pct for Flow: %s\n got %v, want max %v pct failure", f.Name(), lossPct, tolerancePct)
 			} else {
-				t.Logf("Traffic Test Passed!")
+				t.Logf("Traffic Test Passed! for flow %s", f.Name())
 			}
 		} else {
-			if lossPct < 100-tolerancePct {
-				t.Errorf("Traffic is expected to fail %s\n got %v, want 100%% failure", flow.Name(), lossPct)
+			if lossPct < 100-tolerancePct && txPackets > 0 {
+				t.Errorf("Traffic is expected to fail %s\n got %v, want max %v pct failure", f.Name(), lossPct, 100-tolerancePct)
 			} else {
-				t.Logf("Traffic Loss Test Passed!")
+				t.Logf("Traffic Loss Test Passed! for flow %s", f.Name())
 			}
 		}
-	}
-}
 
-func captureTrafficStats(t *testing.T, ate *ondatra.ATEDevice, wantLoss bool) {
-	ap := ate.Port(t, "port1")
-	aic1 := ate.Telemetry().Interface(ap.Name()).Counters()
-	outPkts := aic1.OutPkts().Get(t)
-	fptest.LogYgot(t, "ate:port1 counters", aic1, aic1.Get(t))
-
-	op := ate.Port(t, "port2")
-	aic2 := ate.Telemetry().Interface(op.Name()).Counters()
-	inPkts := aic2.InPkts().Get(t)
-	fptest.LogYgot(t, "ate:port2 counters", aic2, aic2.Get(t))
-
-	lostPkts := inPkts - outPkts
-	t.Logf("Sent Packets: %d, received Packets: %d", outPkts, inPkts)
-	if !wantLoss {
-		if lostPkts > tolerance {
-			t.Logf("Packets received not matching packets sent. Sent: %v, Received: %d", outPkts, inPkts)
-		} else {
-			t.Logf("Traffic Test Passed! Sent: %d, Received: %d", outPkts, inPkts)
-		}
 	}
 }
 
-func sendTraffic(t *testing.T, ate *ondatra.ATEDevice, allFlows []*ondatra.Flow) {
+func sendTraffic(t *testing.T, otg *otg.OTG, c gosnappi.Config) {
 	t.Logf("Starting traffic")
-	ate.Traffic().Start(t, allFlows...)
+	otg.StartTraffic(t)
 	time.Sleep(trafficDuration)
-	ate.Traffic().Stop(t)
 	t.Logf("Stop traffic")
+	otg.StopTraffic(t)
+}
+
+func verifyOtgBgpTelemetry(t *testing.T, otg *otg.OTG, c gosnappi.Config, ipType, state string) {
```

Comment:
verifyOtgBgpTelemetry -> verifyOTGBGPTelemetry because of golang initialism?

Reply:
Done